packages/progress-monitor/src/browser/progress-widget.tsx

```diff
@@ -0,0 +1,95 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { Message } from '@phosphor/messaging';
+import { ReactWidget } from '@theia/core/lib/browser/widgets/react-widget';
+import * as React from 'react';
+import { ProgressService } from './progress-service';
+import { ProgressReport } from './progress-protocol';
+
+@injectable()
+export class ProgressWidget extends ReactWidget {
+
+    static SEARCH_DELAY = 200;
+
+    @inject(ProgressService) protected readonly progressService: ProgressService;
+
+    @postConstruct()
+    protected init() {
+        this.node.classList.add(ProgressWidget.Styles.PROGRESS_WIDGET);
+        this.update();
+        this.node.onclick = e => {
+            e.stopPropagation();
+        };
+        this.progressService.onContributionChanged(() => this.update());
+    }
+
+    protected onActivateRequest(msg: Message) {
+        super.onActivateRequest(msg);
+        this.update();
+    }
+
+    protected render(): React.ReactNode {
+        return this.renderProgressList();
+    }
+
+    protected renderProgressList(): React.ReactNode {
+        const progressList: React.ReactNode[] = [];
+        Array.from(this.progressService.progressItems.values()).forEach(progressItem => {
```

Comment:
maybe `ProgressService.allItems` could already return an array of all items,  instead of exposing the internal map.